### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by DaveZag**
# OOP school library: preserve data
### On this branch we:
- Create files to hold the app data.
- Implement functions to Serialize data before saving it.
- Implement functions to load and save data to and from the computer Storage.
- Implement a function to check file existence(If not found, it creates one).

### CODE REVIEW

File: modules/operations/file_ops.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'json'
+
+def open_file(path)
+  if File.exist?(path)
+    json_string = File.read(path)
+    json_string != '' && json_string != 'null' ? JSON.parse(json_string) : []
+  else
+    File.new(path, 'w')
+    []
+  end
```
Comment (omar-muhamad) [inline modules/operations/file_ops.rb:10]: - [x] As per this [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#:~:text=On%20exit%20from%20your%20app%20save%20all%20the%20data%20(books%2C%20people%2C%20and%20rentals)%20into%20JSON%20files), Please save the people, books, and rentals data on app exit, not just after creating it as you did.
[Hint] In order to do that you should add a method that saves those data and use it when a user exits the app.

File: modules/app.rb [hunk 2]
```diff
@@ -7,9 +7,12 @@ class App
...
   include PersonOperations
 
+  attr_reader :book, :rentals, :person
+
   def initialize
-    @books = []
-    @people = []
+    @books = load_books
+    @people = load_people
+    @rentals = load_rentals
```
Comment (omar-muhamad) [inline modules/app.rb:15]: - [x] After the data is saved and exits the app, When I run the app again there is an error occurred while loading the data. So please fix that.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @DaveZag and @johnkioko 

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [x] Please add your collaborator @johnkioko name and social links in the `Authors` section in the readme file.

image

- [x] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: modules/app.rb (hunk 2 above)
Comment (DaveZag) [reply modules/app.rb:15]: @Omar-Muhamad  Thanks for the feedback. 
The requested changes have been implemented.🙃

**Review by prolajumokeoni — APPROVED** [review]
Hi @DaveZag 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀  
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**